auth/credentials/idtoken/idtoken.go

```diff
@@ -113,13 +113,18 @@ func NewCredentials(opts *Options) (*auth.Credentials, error) {
 	if err := opts.validate(); err != nil {
 		return nil, err
 	}
-	if b := opts.jsonBytes(); b != nil {
-		return credsFromBytes(b, opts)
-	}
-	if metadata.OnGCE() {
+	b := opts.jsonBytes()
+	if b == nil && metadata.OnGCE() {
 		return computeCredentials(opts)
 	}
-	return nil, fmt.Errorf("idtoken: couldn't find any credentials")
+	creds, err := credentials.DetectDefault(&credentials.DetectOptions{
+		Scopes:          defaultScopes,
+		CredentialsJSON: b,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return credsFromBytes(creds.JSON(), opts)
```

Comment:
Can this be refactored a little more to take the whole creds now, that was you don't have to call detect default again later for one of the flows.

Reply:
Updated, PTAL!